### PR TITLE
Register our Payment and Download meta

### PR DESCRIPTION
**Pull request by cklosowski**
#2480

### CODE REVIEW

File: includes/class-edd-register-meta.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		register_meta( 'post', '_edd_payment_customer_id',  array( $this, 'intval_wrapper' ) );
+		register_meta( 'post', '_edd_payment_user_id',      array( $this, 'intval_wrapper' ) );
+		register_meta( 'post', '_edd_payment_user_ip',      'sanitize_text_field' );
+		register_meta( 'post', '_edd_payment_purchase_key', 'sanitize_text_field' );
+		register_meta( 'post', '_edd_payment_total',        'edd_sanitize_amount' );
+		register_meta( 'post', '_edd_payment_mode',         'sanitize_text_field' );
+		register_meta( 'post', '_edd_payment_gateway',      'sanitize_text_field' );
+		register_meta( 'post', '_edd_payment_meta',         array( $this, 'sanitize_array' ) );
+		register_meta( 'post', '_edd_payment_tax',          'edd_sanitize_amount' );
+		register_meta( 'post', '_edd_completed_date',       'sanitize_text_field' );
```
Comment (pippinsplugins) [inline includes/class-edd-register-meta.php]: This is missing a few pieces of meta. Downloads, cart details, and user info.

Comment (pippinsplugins) [inline includes/class-edd-register-meta.php]: @cklosowski Looks like these are still missing.

File: tests/tests-register-meta.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		$this->assertarrayHasKey( 'edd_sanitize_amount', $wp_filter['sanitize_post_meta__edd_payment_tax'][10] );
+		$this->assertarrayHasKey( 'sanitize_text_field', $wp_filter['sanitize_post_meta__edd_completed_date'][10] );
+
+		// Callbacks are part of the object, so just make sure it got registered
+		$this->assertNotEmpty( $wp_filter['sanitize_post_meta__edd_payment_customer_id'][10] );
+		$this->assertNotEmpty( $wp_filter['sanitize_post_meta__edd_payment_user_id'][10] );
+		$this->assertNotEmpty( $wp_filter['sanitize_post_meta__edd_payment_meta'][10] );
+
+	}
+
```
Comment (pippinsplugins) [inline tests/tests-register-meta.php]: Could we add tests that call `update_post_meta()` and passes bad data to `$value` to ensure they're sanitized properly?

File: includes/class-edd-register-meta.php (hunk 1 above)
Comment (cklosowski) [inline includes/class-edd-register-meta.php]: @pippinsplugins those are all subsets of the `payment_meta`, not their own entry into the post meta.

File: tests/tests-register-meta.php (hunk 2 above)
Comment (cklosowski) [inline tests/tests-register-meta.php]: We can, and probably should..will do that and get this updated with latest `release/2.5`

File: includes/class-edd-register-meta.php (hunk 1 above)
Comment (pippinsplugins) [inline includes/class-edd-register-meta.php]: Ah! Right! Carry on

**Comment by cklosowski** [comment]
ok @pippinsplugins ready for another look

**Pull request merged**